src/timeline.js

```diff
@@ -1512,76 +1708,132 @@ export default class Timeline extends React.Component {
-                  forceRedrawFunc={forceRedrawFunc}
-                />
-                {backgroundLayer &&
-                  React.cloneElement(backgroundLayer, {
-                    startDateTimeline: this.getStartDate(),
-                    endDateTimeline: this.getEndDate(),
-                    width: this.state.width,
-                    leftOffset: leftOffset,
-                    height: bodyHeight,
-                    topOffset: timebarHeight,
-                    verticalGridLines: this.state.verticalGridLines
-                  })}
+                  <Timebar
+                    cursorTime={this.getCursor()}
+                    start={this.getStartDate()}
+                    end={this.getEndDate()}
+                    width={this.state.width}
+                    leftOffset={leftOffset}
+                    selectedRanges={this.state.selection}
+                    groupTitleRenderer={groupTitleRenderer}
+                    tableColumns={tableColumns}
+                    groupOffset={groupOffset}
+                    setVerticalGridLines={this.setVerticalGridLines}
+                    {...varTimebarProps}
+                  />
+                  {markers.map(m => (
+                    <Marker
+                      key={m.key}
+                      height={this.state.height}
+                      top={0}
+                      date={0}
+                      shouldUpdate={true}
+                      calculateHorizontalPosition={() => {
+                        return {left: m.left};
+                      }}
+                      className="rct9k-marker-overlay"
+                    />
+                  ))}
+                  <TimelineBody
+                    width={this.state.width}
+                    columnWidth={columnWidth(this.state.width)}
+                    height={bodyHeight}
+                    rowHeight={this.rowHeight}
+                    rowCount={this.state.groups.length}
+                    columnCount={(tableColumns && tableColumns.length > 0 ? tableColumns.length : 1) + 1}
+                    cellRenderer={this.cellRenderer(this.getTimelineWidth(this.state.width))}
+                    grid_ref_callback={this.grid_ref_callback}
+                    shallowUpdateCheck={shallowUpdateCheck}
+                    forceRedrawFunc={forceRedrawFunc}
+                  />
+                  {backgroundLayer &&
+                    React.cloneElement(backgroundLayer, {
+                      startDateTimeline: this.getStartDate(),
+                      endDateTimeline: this.getEndDate(),
+                      width: this.state.width,
+                      leftOffset: leftOffset,
+                      height: bodyHeight,
+                      topOffset: this.state.topOffset + timebarHeight,
+                      verticalGridLines: this.state.verticalGridLines
+                    })}
+                  <div style={{position: 'absolute', right: 1, top: 1}}>{this.renderMenuButton()}</div>
```

Comment:
In general, clasele CSS sunt mai bune din punct de vedere al performantei decart stilurile. Stilurile ar trebui folosite doar atunci cand se calculeaza ceva dinamic.